Shopify/hydrogen
# Store cart ID in cookies

Hey @lordofthecactus! Let me know what you think of this approach.

### WHY are these changes introduced?

When switching from Shopify's Online Store to Hydrogen, existing customer carts are lost unless the merchant updates their liquid theme to use the Storefront API.

Soon, liquid/Storefront API carts will be interoperable - carts created on a traditional liquid storefront can be managed via the Storefront API and vice versa.

This change leverages the new functionality and shows an example of how to switch between Online Store/Hydrogen without losing carts.

### WHAT is this pull request doing?

Updates the Demo Store template to use the same `cart` cookie [set by the traditional Online Store channel](https://screenshot.click/27-52-gldtj-mzwjg.png).

### HOW to test your changes?

1. Enable `cart_repository_online_store_path` on your test store.
2. Create a cart using the traditional Online Store and copy the cart ID from the `cart` cookie.
3. Running Demo Store locally, create the same `cart` cookie.
4. Switch between Hydrogen/Online Store and the cart should remain in sync.

Feel free to use my test store:

```
PUBLIC_STORE_DOMAIN="test-shop-987.myshopify.com"
PUBLIC_STOREFRONT_API_TOKEN="b5f455ac1cea1522791da1ad0e053559"
```

#### Post-merge steps

Will remove most of these docs: https://shopify.dev/docs/custom-storefronts/oxygen/migrate/share-carts

#### Checklist
